sdk/src/message.rs

```diff
@@ -30,35 +30,66 @@ fn compile_instructions(ixs: Vec<Instruction>, keys: &[Pubkey]) -> Vec<CompiledI
         .collect()
 }
 
-/// Return pubkeys referenced by all instructions, with the ones needing signatures first.
-/// If the payer key is provided, it is always placed first in the list of signed keys.
-/// No duplicates and order is preserved.
-fn get_keys(instructions: &[Instruction], payer: Option<&Pubkey>) -> (Vec<Pubkey>, Vec<Pubkey>) {
+/// Return pubkeys referenced by all instructions, with the ones needing signatures first. If the
+/// payer key is provided, it is always placed first in the list of signed keys. Credit-only signed
+/// accounts are placed last in the set of signed accounts. Credit-only unsigned accounts,
+/// including program ids, are placed last in the set. No duplicates and order is preserved.
+fn get_keys(
```

Comment:
Returning a tuple with two items is about as big as we should ever go. Can you create a struct and return that instead? That struct would give names to those `u8`s, which would hopefully would reduce the odds of misuse too.

Reply:
Yeah, I thought about that, with @mvines' recent genesis block example. Thanks for the extra nudge!